extension/background.js

```diff
@@ -0,0 +1,48 @@
+
+
+function showPage(info) {
+  function tabListener(tabId, changeInfo, tab) {
+    if (changeInfo.status == "complete" && tab.url == browser.runtime.getURL("searchdefault.html")) {
+      browser.tabs.sendMessage(tabId, info);
+      browser.tabs.onUpdated.removeListener(tabListener);
+    }
+  }
+  browser.tabs.onUpdated.addListener(tabListener);
+  browser.tabs.create({ url: "searchdefault.html" });
+}
+
+async function runStartup() {
+  // Check for eligible addon on startup
+  let id = await browser.searchDefaults.getEligibleAddonID();
+  console.log(`got back an eligible addon id ${id}`);
+  if (!id) {
+    return;
+  }
+  let info = await browser.management.get(id);
+  showPage(info);
+}
+
+async function initialize() {
+  runStartup();
```

Comment:
Due to the absence of an early return, the logic below (including having a `management` listener) will run unconditionally for all users, even if they don't have any affected extension. Could you add a check and an early return when no eligible add-on is found? That would cover the majority of users.